ir/src/lib.rs

```diff
@@ -146,6 +149,10 @@ impl AirIR {
         &self.constants
     }
 
+    pub fn segment_width(&self) -> &Vec<u16> {
+        &self.segment_width
+    }
+
```

Comment:
nit: I think I'd prefer `segment_widths` instead, to be clearer that it has the width of all the segments.